### ISSUE TITLE
Feature Request: Integrate Microsoft.Extensions.AI into Google.Cloud.AIPlatform

### ISSUE THREAD

**Issue by Krzysztof318** [googleapis/google-cloud-dotnet#13815]
## Feature Request: Integrate Microsoft.Extensions.AI with Google.Cloud.AIPlatform

**Is your feature request related to a problem? Please describe.**

While Google Cloud SDK for .NET and specifically the `Google.Cloud.AIPlatform` library provide robust tools for interacting with Google AI services, the current implementations lack a unified abstraction layer compatible with Microsoft.Extensions.AI.

**Describe the solution you'd like**

I propose integrating Microsoft.Extensions.AI with `Google.Cloud.AIPlatform` (v1/v1beta). This library offers a unified API abstraction for AI services, similar to logging and dependency injection abstractions in .NET. By leveraging Microsoft.Extensions.AI, we can achieve:

* Aligned development experience: Developers can utilize familiar abstractions, streamlining the integration of Google AI services.
* Enhanced flexibility: Developers can easily switch between Google AI services and other providers with minimal code changes.
* Improved code maintainability: A consistent abstraction layer promotes the creation of modular and reusable AI components.

**Additional context**

* Microsoft.Extensions.AI announcement: [Microsoft DevBlogs](https://devblogs.microsoft.com/dotnet/introducing-microsoft-extensions-ai-preview/)
* Real-world example: Implementation of Microsoft.Extensions.AI in AWS SDK for .NET [AWS Pull Request](https://github.com/aws/aws-sdk-net/pull/3545)
* Initial PR as a concept: #13814 

## Questions to the Google Team

1. **Potential challenges:** What are the potential challenges and trade-offs of adopting this approach in your repo?
2. **Preferred implementation:** What would be the preferred approach for implementing this integration? Would it involve creating new project or extend existing AIPlatform projects?
It would be best option to be consistent with Microsoft.Extensions.AI idea but I am open to suggestions from your side.

I'm eager to discuss this proposal further and go work on pull request when we will agree on the details.

I believe integrating Microsoft.Extensions.AI with `Google.Cloud.AIPlatform` would significantly benefit the .NET developer community and foster a more unified approach to AI development on Google Cloud.

cc: @stephentoub

**Comment by amanda-tarafa** [comment]
@Krzysztof318 thanks! We need to figure out a few things internally, so we need a couple of days. I'll keep this issue updated.

**Comment by amanda-tarafa** [comment]
Just here to say, we are still looking into this.

**Comment by Krzysztof318** [comment]
@amanda-tarafa Hi, any updates?

**Comment by amanda-tarafa** [comment]
No, I'm sorry. It's unlikely we'll have any updates before end of year.

I will share any information I can as soon as I have it.

**Comment by Krzysztof318** [comment]
Hello in new year, I just remind about this :)

**Comment by amanda-tarafa** [comment]
Hello, thanks for the reminder. We have not forgotten about it!

**Comment by Krzysztof318** [comment]
Hello @amanda-tarafa I undestarnd you are currently busy with much more important tasks but could provide approximate date about starting this feature? I will glad to write entire code for that but first I have to know yours expectations about this.

**Comment by amanda-tarafa** [comment]
@Krzysztof318 I'm really sorry I cannot share a timeline for this. But this is something we are still looking closely into, and not in the least forgotten. As soon as I have details I can share I will update the issue.

**Comment by stephentoub** [comment]
> [@Krzysztof318](https://github.com/Krzysztof318) We'll be working on this in the next coming weeks to produce a prototype. Because we'll be providing ongoing maintenance, we'll do the main work ourselves, but we are happy to accept contributions to fill in the gaps, and also to have your review on PRs.

This is great. Please let me know if/how I can help.

**Comment by Krzysztof318** [comment]
@amanda-tarafa Thank you for your response. The most important thing is that we are moving forward with the project. I am looking forward to the prototype.

**Comment by luisquintanilla** [comment]
Hi @amanda-tarafa, 

I'm on the team working on Microsoft.Extensions.AI. Just checking in if there'd been any progress on this issue.

As Stephen mentioned, let us know if / how we can help here.

**Comment by amanda-tarafa** [comment]
Nothing to share at the moment. We'll update this issue as soon as we are able.

**Comment by stephentoub** [comment]
Would it be helpful if we submitted a draft PR showing what it would look like, and iterate from there?

**Comment by amanda-tarafa** [comment]
@stephentoub Would you and others be up for a chat about this and some other touchpoints we share? Thinking also of https://github.com/dotnet/aspnetcore/issues/61817. Let me know and I can look into setting things up.

@Krzysztof318 and others requesting this integration, we are still looking into supporting this as soon as we are able.

**Comment by jeffhandley** [comment]
@amanda-tarafa I sent you a message through LinkedIn so that we can connect and set up a meeting about Microsoft.Extensions.AI and ASP.NET auth. /cc @jskeet.

**Comment by tapizquent-gail** [comment]
Any updates on this?

**Comment by AlexanderMagdysyuk** [comment]
> [@amanda-tarafa](https://github.com/amanda-tarafa) I sent you a message through LinkedIn so that we can connect and set up a meeting about Microsoft.Extensions.AI and ASP.NET auth. /cc [@jskeet](https://github.com/jskeet).

@jeffhandley - Can somebody share results of this meeting please?

**Comment by amanda-tarafa** [comment]
Quick update: this is still on our backlog, but with no current ETA. It's possible that we can get it in for Q4, but that's just a possibility.

I'm sorry for the wait, it's just a matter of many competing priorities at the moment.

**Comment by stephentoub** [comment]
I will submit a PR for this once https://github.com/googleapis/google-cloud-dotnet/pull/15131 is merged.

**Comment by amanda-tarafa** [comment]
> I will submit a PR for this once [#15131](https://github.com/googleapis/google-cloud-dotnet/pull/15131) is merged.

#15131 is merged, thanks!

**Comment by amanda-tarafa** [comment]
I'm reopening, we still have some tweaking to do on our side re project generation, etc.

**Comment by amanda-tarafa** [comment]
I'm working on some of the tweaks, aiming to release a beta at some point next week.

**Comment by amanda-tarafa** [comment]
#15310 is first follow up and another one coming after.

Expectations are that we can release this week, but that's not a hard commitment.

**Comment by amanda-tarafa** [comment]
Quick update: It seems like embeddings should be handled by the `predict` endpoint and not the `embedContent` endpoint, see [Get Text Embeddings](https://docs.cloud.google.com/vertex-ai/generative-ai/docs/embeddings/get-text-embeddings) and [Get Multimodal Embeddings](https://docs.cloud.google.com/vertex-ai/generative-ai/docs/embeddings/get-multimodal-embeddings).

@stephentoub any particular reason you chose `embedContent`? I have the code for this already but I'm waiting for confirmation etc.

**Comment by stephentoub** [comment]
> @stephentoub any particular reason you chose embedContent

I didn't see other embedding APIs on predictionserviceclient (and still don't actually... what's it called?) . If there's a better api to delegate to, though, great.

**Comment by stephentoub** [comment]
> It seems like embeddings should be handled by the predict endpoint

Oh, by predict? If you look in the commit history, that was previously there. My last pr changed it to EmbedContent when those methods were added.

https://github.com/googleapis/google-cloud-dotnet/commit/043a61cc89ec6406a726e518c792af084c200064#diff-a254591d0d198208885a3ae305c3c2d3c94f42334e7119aac287946f06003a1d

If we don't want folks using EmbedContent for the vastly most popular form of embedding, there's a bad usability issue there with it being exposed. But you can just revert back to the previous implementation, shouldn't need new code.

**Comment by amanda-tarafa** [comment]
> If we don't want folks using EmbedContent for the vastly most popular form of embedding, there's a bad usability issue there with it being exposed.

100% agree, on the libraries side we don't have much control over what each API team exposes. I'm talking to them but for now I'll merge your revert PR. Thanks!

**Comment by amanda-tarafa** [comment]
Initial beta has been published: https://www.nuget.org/packages/Google.Cloud.VertexAI.Extensions/1.0.0-beta01

I'll close this issue now, we can deal with improvements/bugs on a case by case basis. Please report any issues on this repo which is the fastest way to bring them to our attention.

**Comment by stephentoub** [comment]
Thanks, @amanda-tarafa.
